### PR TITLE
codegen: Make comments indentation-aware.

### PR DESCRIPTION
**Pull request by emilio**
This commit moves comment processing to a central place (well, two, because of
field docs, but that's fine).

Also, it makes comments indentation aware, so multiline comments don't appear
garbled.

Finally, it also fixes an out-of-bounds panic when processing an empty multiline
comment.

### CODE REVIEW

File: src/ir/item.rs [hunk 1]
```diff
@@ -480,9 +480,35 @@ impl Item {
...
+        }
+
+        self.ancestors(ctx)
+            .filter(|id| {
+                ctx.resolve_item(*id).as_module().map_or(false, |module| {
+                    !module.is_inline() ||
+                        ctx.options().conservative_inline_namespaces
+                })
+            })
+            .fold(1, |i, _| i + 1)
```
Comment (fitzgen) [inline src/ir/item.rs]: Isn't this fold just `<iterator>.count() + 1`? Better not to make readers verify that the fold is just counting, when there's a std method for counting.

File: src/ir/comment.rs [hunk 2]
```diff
@@ -79,16 +103,16 @@ mod test {
...
     }
 
     #[test]
     fn processes_multi_lines_correctly() {
-        assert_eq!(preprocess("/** hello \n * world \n * foo \n */".to_owned()),
+        assert_eq!(preprocess("/** hello \n * world \n * foo \n */", 0),
                    "/// hello\n/// world\n/// foo");
 
-        assert_eq!(preprocess("/**\nhello\n*world\n*foo\n*/".to_owned()),
+        assert_eq!(preprocess("/**\nhello\n*world\n*foo\n*/", 0),
```
Comment (fitzgen) [inline src/ir/comment.rs]: Would be nice to have a unit test with nonzero indent for comment preprocessing.

File: src/ir/item.rs (hunk 1 above)
Comment (emilio) [inline src/ir/item.rs]: I just moved code around, but sure :)

File: src/ir/comment.rs (hunk 2 above)
Comment (emilio) [inline src/ir/comment.rs]: I'm not a super-fan of unit tests, but will do

Comment (emilio) [inline src/ir/comment.rs]: (err, for stuff that can be integration-tested, I mean)

**Pull request merged**